keras_core/layers/preprocessing/random_rotation.py

```diff
@@ -76,52 +68,161 @@ class RandomRotation(Layer):
-            name=name,
-            **kwargs,
-        )
         self.supports_jit = False
         self._convert_input_args = False
         self._allow_non_tensor_positional_args = True
 
+    def _set_value_range(self, value_range):
+        if not isinstance(value_range, (tuple, list)):
+            raise ValueError(
+                self.value_range_VALIDATION_ERROR
+                + f"Received: value_range={value_range}"
+            )
+        if len(value_range) != 2:
+            raise ValueError(
+                self.value_range_VALIDATION_ERROR
+                + f"Received: value_range={value_range}"
+            )
+        self.value_range = sorted(value_range)
+
+    def _set_factor(self, factor):
+        if isinstance(factor, (tuple, list)):
+            if len(factor) != 2:
+                raise ValueError(
+                    self._FACTOR_VALIDATION_ERROR + f"Received: factor={factor}"
+                )
+            self._check_factor_range(factor[0])
+            self._check_factor_range(factor[1])
+            self._factor = sorted(factor)
+        elif isinstance(factor, (int, float)):
+            self._check_factor_range(factor)
+            factor = abs(factor)
+            self._factor = [-factor, factor]
+        else:
+            raise ValueError(
+                self._FACTOR_VALIDATION_ERROR + f"Received: factor={factor}"
+            )
+
+    def _check_factor_range(self, input_number):
+        if input_number > 1.0 or input_number < -1.0:
+            raise ValueError(
+                self._FACTOR_VALIDATION_ERROR
+                + f"Received: input_number={input_number}"
+            )
+
+    """
+    Assume an angle ø, then rotation matrix is defined by 
+    | cos(ø)   -sin(ø)  x_offset |
+    | sin(ø)    cos(ø)  y_offset |
+    |   0         0         1    |
+    
+    This function is returning the 8 elements barring the final 1 as a 1D array
+    """
+
+    def get_rotation_matrix(self, inputs):
+        input_shape = len(inputs.shape)
+        if input_shape == 4:
+            if self.data_format == "channels_last":
+                batch_size = inputs.shape[0]
+                image_height = inputs.shape[1]
+                image_width = inputs.shape[2]
+            else:
+                batch_size = inputs.shape[1]
+                image_height = inputs.shape[2]
+                image_width = inputs.shape[3]
+        else:
+            batch_size = 1
+            if self.data_format == "channels_last":
+                image_height = inputs.shape[0]
+                image_width = inputs.shape[1]
+            else:
+                image_height = inputs.shape[1]
+                image_width = inputs.shape[2]
+
+        lower = self._factor[0] * 2.0 * backend.convert_to_tensor(np.pi)
+        upper = self._factor[1] * 2.0 * backend.convert_to_tensor(np.pi)
+
+        angle = backend.random.uniform(shape=(batch_size,), minval=lower, maxval=upper, dtype=np.float32)
```

Comment:
You need to pass a seed generator, otherwise you get the same value every time.

Use `seed_generator = self._get_seed_generator(self.backend._backend)` then `seed=seed_generator`